auth/server/base/src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -2023,6 +2024,18 @@ AuthorizedAuthenticationState doAuthorization(final boolean requireLoginPermissi
 
             SecurityIdentity authorizedIdentity = Assert.assertNotNull(domain.transform(new SecurityIdentity(domain, authenticationPrincipal, realmInfo, authorizationIdentity, domain.getCategoryRoleMappers(), IdentityCredentials.NONE, IdentityCredentials.NONE)));
             authorizedIdentity = authorizedIdentity.withPublicCredentials(publicCredentials).withPrivateCredentials(privateCredentials);
+            Set<Credential> privateCredentialsFromRealmIdentity = realmIdentity.getPrivateCredentials();
+            if (privateCredentialsFromRealmIdentity != null) {
+                for (Credential credential : privateCredentialsFromRealmIdentity) {
+                    authorizedIdentity = authorizedIdentity.withPrivateCredential(credential);
```

Comment:
Each iteration of the for loop will result in a new SecurityIdentity being created and one more being left for GC.  Could this be reduced by converting to an IdentityCredentials first?